viam-cartographer.go

```diff
@@ -133,7 +128,8 @@ func New(
 		return nil, err
 	}
 
-	cancelCtx, cancelFunc := context.WithCancel(ctx)
+	// Need to pass in a long-lived context because ctx is short-lived
```

Comment:
```suggestion
	// Need to pass in a long-lived context because ctx is short-lived
	// To be re-evaluated in https://viam.atlassian.net/browse/RSDK-2586
```